src/NifFile.cpp

```diff
@@ -1409,8 +1407,8 @@ OptResult NifFile::OptimizeFor(OptOptions& options) {
 				if (!hasTangents && bsOptShape->HasTangents())
 					result.shapesTangentsAdded.push_back(shapeName);
 
-				hdr.ReplaceBlock(GetBlockID(shape), bsOptShape);
-				UpdateSkinPartitions(bsOptShape);
+				hdr.ReplaceBlock(GetBlockID(shape), std::unique_ptr<NiObject>(std::move(bsOptShape)));
+				UpdateSkinPartitions(bsOptShape.get());
```

Comment:
@Guekka Commit 3e2e36b is breaking optimization because bsOptShape ptr is moved away before calling UpdateSkinPartitions.